### PR TITLE
backport fixes for MariaDB 11.4

### PR DESCRIPTION
**Pull request by tuxmaster5000**
Fix for https://github.com/bareos/bareos/issues/2091
Tested on MariaDB 11.4 and 10.11 on Rocky/RHEL 9.5.
Because 11.4 has removed the mysql commands.
#### Please check

- [X] Short description and the purpose of this PR is present _above this paragraph_
- [ ] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)
- [Boy Scout Rule](https://docs.bareos.org/DeveloperGuide/generaldevel.html#boy-scout-rule)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [ ] Is the PR title usable as CHANGELOG entry?
- [ ] Purpose of the PR is understood
- [ ] Commit descriptions are understandable and well formatted
- [ ] Required backport PRs have been created
- [ ] Correct milestone is set

##### Source code quality
- [ ] Source code changes are understandable
- [ ] Variable and function names are meaningful
- [ ] Code comments are correct (logically and spelling)
- [ ] Required documentation changes are present and part of the PR

##### Tests
- [ ] Decision taken that a test is required (if not, then remove this paragraph)
- [ ] The choice of the type of test (unit test or systemtest) is reasonable
- [ ] Testname matches exactly what is being tested
- [ ] On a fail, output of the test leads quickly to the origin of the fault

### CODE REVIEW

**Comment by bruno-at-bareos** [comment]
Thanks for the proposal, to make it there's a few things to fix.

First as stated in our development documentation, you should force yourself to use cherry-pick -x when doing commit backport

Second point actually our CI is failing tests on different plateform like F40 with mariadb 10.x (as this one should still work of course).
```
Using Catalog "MyCatalog"
Job queued. JobId=4
@tee /var/lib/jenkins/workspace/bareos_PR-2093@2/BUILD/bareos/cmake-build/systemtests/tests/py3plug-fd-mariabackup/tmp/log2.out
wait
You have messages.
shutdown MySQL server waiting for pid 47910 (/usr/sbin/mysqld --defaults-file=mysqldefaults) . OK
ls: cannot access '/var/lib/jenkins/workspace/bareos_PR-2093@2/BUILD/bareos/cmake-build/systemtests/tests/py3plug-fd-mariabackup/tmp/bareos-restores/_mariabackup/': No such file or directory
exit(2) is called. Set test to failure and end test.
Stopping the Bareos File daemon: OK
Stopping the Bareos Storage daemon: .... OK
...
```

**Comment by tuxmaster5000** [comment]
Without the test data, it is hard to say what goes wrong. Because when the URL is called up, all you get is: Access denied.
But from a distance, it looks like the test is not creating the restore directory.

**Comment by bruno-at-bareos** [comment]
On vendredi, 3 janvier 2025 13.35:22 h heure normale d’Europe centrale 
tuxmaster5000 wrote:
> Without the test data, it is hard to say what goes wrong. Because when the
> URL is called up, all you get is: Access denied. But from a distance, it
> looks like the test is not creating the restore directory.

That's mainly because you didn't create a backport and use git cherry-pick -x 
#commit from the original PR (which should contain almost everything)
Once that's done, you might still need to adapt a few things, which needs to 
be documented here in the description.

teaser the `devtool/pr-tool` is a nice helper for that task.

**Comment by tuxmaster5000** [comment]
Yesterday I tested the restore again using MariaDB 11.4 + 10.11 on RHEL-9 and it works. So I don't have an Idea, what goes wrong. And without access to the raw test logs I can't do anything. :(

**Comment by pstorz** [comment]
You have changed the subdirectory name from "_mariabackup" to "_mariadb-backup" but not in the 
systemtest:

cannot access 
***@***.***/BUILD/bareos/cmake-build/systemtests/tests/py3plug-fd-mariabackup/tmp/bareos-restores/_mariabackup/': 
No such file or directory
exit(2) is called. Set test to failure and end test.


Am 04.01.25 um 09:17 schrieb tuxmaster5000:

**Comment by tuxmaster5000** [comment]
Hopefully, I have found all lines.

**Comment by bruno-at-bareos** [comment]
Hello, thanks for the effort, but as I mentioned in earlier comment, your backport needs to followup the format we use (as state in documentation https://docs.bareos.org/DeveloperGuide/gitworkflow.html#releases-and-backporting)

Essentially you need to backport with `git cherry-pick -x` all the commit needed to make it work here. As we and (you) will need to maintain those changes, we can't decently accept copied and pasted random code without a traceability from where (and the why) it comes from.

Also the tool present in `devtool/pr-tools` can help in creating the backport 

```devtools/pr-tool backport create --into bareos-23 --all 1775

**Comment by bruno-at-bareos** [comment]
Hello this is a kind request, do you want to recreate the backport with the rules?

**Comment by tuxmaster5000** [comment]
As I can usually only work on it in my free time, progress is slow. But yes, that is planned.

**Comment by bruno-at-bareos** [comment]
Hello, I don't know why you didn't get any error of the tool because when I tried you command the message is quite clear

```
devtools/pr-tool backport create --into bareos-23 --all 1775
CRITICAL: backport: Cannot get information for PR 1775. Does it exist?
```

In fact you trie to use #1775 which is the issue number, what we want to backport is the the Pull request #1835 which has fixed that issue. (Yeah github is using the same number for issue and pr, which might be confusing)

So the right call looks like
```
devtools/pr-tool backport create --into bareos-23 --all 1835
```

Which will give you the following, and then you will have to fix the commits 
and you will be placed automatically in a new branch called `backport/bareos-23/fix-1775-add-support-mariadb-11+`

```
Auto-merging core/cmake/BareosFindPrograms.cmake
CONFLICT (content): Merge conflict in core/cmake/BareosFindPrograms.cmake
Auto-merging systemtests/environment.in
Auto-merging systemtests/tests/py3plug-fd-mariabackup/CMakeLists.txt
CONFLICT (content): Merge conflict in systemtests/tests/py3plug-fd-mariabackup/CMakeLists.txt
CONFLICT (modify/delete): systemtests/tests/py3plug-fd-mariabackup/testrunner deleted in e9bf76432 (plugin: fd mariabackup add support for mariadb 11+) and modified in HEAD.  Version HEAD of systemtests/tests/py3plug-fd-mariabackup/testrunner left in tree.
error: could not apply e9bf76432... plugin: fd mariabackup add support for mariadb 11+
hint: After resolving the conflicts, mark them with
hint: "git add/rm <pathspec>", then run
hint: "git cherry-pick --continue".
hint: You can instead skip this commit with "git cherry-pick --skip".
hint: To abort and get back to the state before "git cherry-pick",
hint: run "git cherry-pick --abort".
hint: Disable this message with "git config set advice.mergeConflict false"
Recorded preimage for 'core/cmake/BareosFindPrograms.cmake'
Recorded preimage for 'systemtests/tests/py3plug-fd-mariabackup/CMakeLists.txt'
```

Basically the things to be fixed are
```
## backport/bareos-23/fix-1775-add-support-mariadb-11+...origin/backport/bareos-23/fix-1775-add-support-mariadb-11+ [gone]
UU core/cmake/BareosFindPrograms.cmake
M  core/src/plugins/filed/python/mariabackup/bareos-fd-mariabackup.py
M  systemtests/environment.in
UU systemtests/tests/py3plug-fd-mariabackup/CMakeLists.txt
M  systemtests/tests/py3plug-fd-mariabackup/etc/bareos/bareos-dir.d/fileset/MariabackupTest.conf.in
UD systemtests/tests/py3plug-fd-mariabackup/testrunner
A  systemtests/tests/py3plug-fd-mariabackup/testrunner-default
```

Hope this help to continue the work.

**Comment by bruno-at-bareos** [comment]
@tuxmaster5000 have you been able to try the last comment tricks ?
Just a kind reminder if done it might get in next release.

**Comment by tuxmaster5000** [comment]
I don't know how to fix the tests.
But I have seen another problem in the module on bareos 24. So I think the patch will also be affected.
So 2 parts are open, fix the tests and  wait for the needed changed of the add-on in bareos 24 (because it must back ported also)

**Comment by bruno-at-bareos** [comment]
On mercredi, 5 février 2025 14.07:09 h heure normale d’Europe centrale 
tuxmaster5000 wrote:
> I don't know how to fix the tests.
> But I have seen another problem in the module on bareos 24. So I think the
> patch will also be affected. So 2 parts are open, fix the tests and  wait
> for the needed changed of the add-on in bareos 24 (because it must back
> ported also)

It might take time before getting the patch for the other issue.
Then if you won't work on this PR before getting the other fixes, I will place 
it on nobuild and onhold so we free up resources on the CD/CI ..

Thanks for the feedback

**Comment by tuxmaster5000** [comment]
That's a good plan. We are now considering how far it makes sense to continue building the 23 series or whether we should go straight to the 24 series.  What would really help everyone would be if the logs were collected as artefacts and sent back to the PR. With other Git servers this works well, so that the developers don't need access to the CI service, but can read the logs. This will save much time on all sites.

**Comment by bruno-at-bareos** [comment]
Hi @tuxmaster5000 what was the result of your research, will you still build 23 and as such will continue to pursue the effort on this PR, or we can close it?
The CI log access is still an ongoing task here, we're really hoping to be able to do something on that subject this year.

**Comment by tuxmaster5000** [comment]
Hi @bruno-at-bareos without access to the build logs I can't do anything.
On my systems the patch will work, but when I try to open the url for the build log I only get an 403.

**Comment by bruno-at-bareos** [comment]
> Hi @bruno-at-bareos without access to the build logs I can't do anything. On my systems the patch will work, but when I try to open the url for the build log I only get an 403.

You're patch won't be go in as it is, if there's a fix, it has to be a backport of code fix like describe in https://github.com/bareos/bareos/pull/2093#issuecomment-2601858988

**Comment by tuxmaster5000** [comment]
Working blindly makes it impossible.
Because here on the Github page, no conflicts are displayed.
Otherwise, there is always a corresponding message directly from GitHub's PR.

**Comment by bruno-at-bareos** [comment]
In a week (30th April 2025) without a proper proposal I will close this.
